src/Type/Symfony/BrowserKitAssertionTraitReturnTypeExtension.php

```diff
@@ -0,0 +1,68 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Symfony;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use PHPStan\Analyser\Scope;
+use PHPStan\Type\ExpressionTypeResolverExtension;
+use PHPStan\Type\NullType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\UnionType;
+use function count;
+
+final class BrowserKitAssertionTraitReturnTypeExtension implements ExpressionTypeResolverExtension
+{
+
+	private const TRAIT_NAME = 'Symfony\Bundle\FrameworkBundle\Test\BrowserKitAssertionsTrait';
+	private const TRAIT_METHOD_NAME = 'getClient';
+
+	public function getType(Expr $expr, Scope $scope): ?Type
+	{
+		if ($this->isSupported($expr, $scope)) {
+			$args = $expr->getArgs();
+			if (count($args) > 0) {
+				return TypeCombinator::intersect(
+					$scope->getType($args[0]->value),
+					new UnionType([
+						new ObjectType('Symfony\Component\BrowserKit\AbstractBrowser'),
+						new NullType(),
+					]),
+				);
+			}
+
+			return new ObjectType('Symfony\Component\BrowserKit\AbstractBrowser');
+		}
+
+		return null;
+	}
+
+	/**
+	 * @phpstan-assert-if-true =MethodCall $expr
+	 */
+	private function isSupported(Expr $expr, Scope $scope): bool
+	{
+		if (!($expr instanceof MethodCall) || !($expr->name instanceof Identifier) || $expr->name->name !== self::TRAIT_METHOD_NAME) {
```

Comment:
I don't understand the logic. You're interested in the method name and you're interested where we're calling it from.

But you're not interested on what we're calling the method. This extension might be influencing a total coincidence that's nothing to do with `BrowserKitAssertionsTrait::getClient()`.

We're in a class, the class has `getClient` method. And the method is declared in BrowserKitAssertionsTrait trait. But the method we're still calling might be unrelated.

My hunch is telling me you're no checking `$scope->getMethodReflection($scope->getType($expr->var), $expr->name->name)` but you should be.

Also, method names are case-insensitive, so toLowerString() needs to happen.